platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/AnnotationContainer.kt

```diff
@@ -41,17 +44,23 @@ class KAnnotationContainer(
         }
     }
 
+    @UiThread
     fun update(annotation: KAnnotation<*>) {
         managers[annotation.key()]?.updateSource()
-        if (annotation is KSymbol) annotation.icon?.let { style?.addImage(it.image.toString(), it.image) }
+        if (annotation is Symbol) annotation.icon?.let { style?.addImage(it.image.toString(), it.image) }
+        if (annotation is Line) annotation.pattern?.let { style?.addImage(it.toString(), it) }
+        if (annotation is Fill) annotation.pattern?.let { style?.addImage(it.toString(), it) }
```

Comment:
> Are you iterating over all the markers to handle the drag event?

@louwers No, like previously, we are only calling `AnnotationManager.queryMapForFeatures` one time to get the annotation to be dragged when dragging begins. What I considered could be a bottleneck is that we are re-adding the icon image to the map every update. However, commenting out the lines marked above does not noticeably improve the situation.

Reply:
Was the performance this bad before too?

Let's have a chat on Slack when you have time.